runtime/parachains/src/router/ump.rs

```diff
@@ -15,15 +15,34 @@
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
 use super::{Trait, Module, Store};
-use crate::configuration::HostConfiguration;
+use crate::configuration::{self, HostConfiguration};
 use sp_std::prelude::*;
-use frame_support::{StorageMap, StorageValue, weights::Weight};
+use sp_std::collections::{btree_map::BTreeMap, vec_deque::VecDeque};
+use frame_support::{StorageMap, StorageValue, weights::Weight, traits::Get};
 use primitives::v1::{Id as ParaId, UpwardMessage};
 
+pub trait UmpSink {
```

Comment:
some docs on this please :)

Reply:
Added, and realized that we still haven't figured out the limits on the `CandidateCommitments`

Reply:
which limits?

Reply:
basically, currently, at the moment `CandidateCommitments` is mostly of constant size with exception of messages. HRMP and UMP outbound messages inflate the candidate commitments. HRMP has some upper bound of size: a parachain can open a limited number of channels and the sizes of messages are limited as well. UMP has only a limit on number of messages and these messages can be of any size. 

Therefore, if I am not mistaken, this means that `CandidateCommitments` can be inflated as well. Not sure, which would be the next limit (perhaps networking? or do we have a special limit for `CandidateCommitments`?), but I think we are better off introducing a dedicated limit or limits, that would effectively limit the overall size taken in candidate commitments. 

Now, apart from the size concern there is also the weight.  But now after dekindification (#1702, also tackled in this PR), all the messages are byte blobs anyway, so UMP delegates this to a higher level. UMP dispatching aside, we need to weigh all this machinery that propels it. 

I had this in mind and I was planning to discuss this on the second call with you and Al, but that never happened and then I forgot until I was writing [these docs](https://github.com/paritytech/polkadot/pull/1679/commits/0d1835c7a2cef750828a14f2caf2a0793191c4bc#diff-65cfe3153bc3f3af755d35c872a99d790a940efd54223693a492f4bcf7b2a447R26).

Reply:
moved this to issue https://github.com/paritytech/polkadot/issues/1869. Feel free to reply there